src/VisualStudio/Core/Def/Implementation/Library/FindResults/TreeItems/MetadataDefinitionTreeItem.cs

```diff
@@ -36,7 +35,17 @@ public MetadataDefinitionTreeItem(ISymbol definition, ushort glyphIndex)
 
         public override int GoToSource()
         {
-            return VSConstants.E_NOTIMPL;
+            var resolution = _symbolKey.Resolve(_workspace.CurrentSolution.GetCompilationAsync(_referencingProjectId, CancellationToken.None).Result);
```

Comment:
@KirillOsenkov and I spent a late night debugging a few years ago to determine that popping a threaded wait dialog from here can potentially cause the Find Results window to hide again if pressing Shift+F12 would cause it slide out.